history.go

```diff
@@ -46,24 +69,24 @@ func (h *History) LookUpInEpoch(key string, ep int64) (MerkleNode, *Authenticati
 	return n, ap, nil
 }
 
-func (h *History) GetSTR(ep int64) *SignedTreeRoot {
-	pointer := h.str
-	for pointer.epoch > ep && pointer != nil {
-		if pointer.prev == nil {
+func (h *History) GetSTR(ep Epoch) *SignedTreeRoot {
+	pointer := h.hashChain[h.currentSTR.epoch]
+	for pointer.epoch > ep && h.hashChain[pointer.epoch] != nil {
```

Comment:
I'm wondering if such a check would also be needed for the current implementation. I think using ``epochs`` could be a bit cleaner, but either implementation works for me. Maybe others have stronger opinions one way or the other.